### ISSUE TITLE
HomematicIP-Devices offline with 2.4.0 Snapshot 1330

### ISSUE THREAD

**Issue by michaeljoos72** [eclipse/smarthome#6053]
Hi

Not sure if I'm the only one, but the Update from openHAB 2.4 Snapshot 1328 to Snapshot 1330 did kill the connection to all my HomematicIP-Devices. I did not change anything else. CCU2 is on latest FW. CCU & all non-IP-Devices are online and working.

https://community.openhab.org/t/homematicip-devices-offline-with-2-4-0-snapshot-1330/49427

Trace-Log says:

```
2018-08-16 06:09:28.782 [TRACE] [nal.communicator.client.BinRpcClient] - Client BinRpcRequest:
getParamsetDescription()
MEQ0737879
VALUES

2018-08-16 06:09:28.794 [TRACE] [al.communicator.client.SocketHandler] - Returning socket for port 2001
2018-08-16 06:09:28.800 [TRACE] [nal.communicator.client.BinRpcClient] - Client BinRpcResponse:
{
_	faultCode=-3_
_	faultString=Unknown paramset_
}

2018-08-16 06:09:28.803 [DEBUG] [tic.internal.communicator.CcuGateway] - RpcMessage RPC failure (-3 Unknown paramset), fetching metadata with TclRega script for device: MEQ0737879, channel: -1, paramset: VALUES
```

@gerrieg Did we have a change in the Binding?

Thanks
Michael

[openhab.log](https://github.com/eclipse/smarthome/files/2292584/openhab.log)

**Comment by gerrieg** [comment]
I haven't changed anything, I'm on vacation. I can't check until the middle of next week.

**Comment by colognebroker** [comment]
I have the same Problem NO HM IP devices are shown !
Not in inbox and not in Binding

**Comment by gerrieg** [comment]
This PR breakes the HmIP devices: https://github.com/eclipse/smarthome/pull/5963

@FStolte 
I'm not sure what this PR is about. What is the -1 channel for?
HmIP Devices must have a valid channel definition like address:channel for the rpc calls

**Comment by tomhoefer** [comment]
@FStolte can you pls check this bug?

**Comment by FStolte** [comment]
@gerrieg The PR you mention integrates support for the logical devices reported by the gateway that do not map to a channel. For one physical device, the gateway reports channels, or "logical devices" like this: "MEQ008963:2". However, there is also always a logical device with an address like "MEQ008963" that provides a MASTER Paramset which was not mapped to a channel by the binding previously. By supporting this Paramset in the binding, we get more configuration parameters, e.g. for the homematic thermostat.

To avoid bigger changes, I just mapped this logical device to a new channel with number -1, so that the infrastructure of the binding to deal with the datapoints of that Paramset could easily be reused.
Unfortunately, I did not check this new implementation with a HmIP device, and therefore did not discover these issues. I can see two problems:
1. I did not specify that the -1 channel only has a MASTER Paramset, which is why the binding is trying to get the ParamsetDescription for the VALUES Paramset, which fails for HmIP.
2. I only implemented the mapping of the -1 Channel in the RpcClient, but from the logs I can see that the generated TclRegaScript contains the channel "-1", which it should not. There are other parts of the binding which might also behave unexpectedly when encountering this channel value.

I can think of a simple solution for the first problem: In the `addChannelDatapoints` method of the RpcClient, we must ensure that we don't try to request the VALUES Paramset for channel -1. This should solve the HmIP problems.
However, if there is an `UnknownParameterSetException` for some other reason, the TclRegaScript would still be called with a channel value of -1. Looking at the script in the logs, I think that just passing an empty string for the channel number will not result in the desired behavior, so bigger changes to the script would be necessary to also support the "-1 channel".
It might have been a bit naive to use this "-1 channel" shortcut, and I think to avoid other problems that have potentially been introduced by this change, I need to create a proper integration for these additional logical devices that does not rely on the HmChannels.

TL;DR: I'm pretty sure I know how to fix this (already working on it), although to avoid other potential problems in the future, bigger changes might become necessary.

**Comment by colognebroker** [comment]
NO Homematic IP devices with OH stable 2.4 ?
also when i try to add Devices manually Paper UI did not offer Homematic IP Components
It seems that OH did not know anything about HIP devices

**Comment by colognebroker** [comment]
On my old 2.2 Configuration everything is fine
![oh 2 2]
<img width="995" alt="oh 2 4" src="https://user-images.githubusercontent.com/23420078/44523967-2810b680-a6dc-11e8-9c8c-3426f1a3da9f.png">
![oh 2 2](https://user-images.githubusercontent.com/23420078/44523978-3068f180-a6dc-11e8-9727-47f1c18c4b70.png)

**Comment by kaikreuzer** [comment]
@michaeljoos72 FYI, I just built a new ESH stable and started an openHAB distro build, so that the next distro snapshot (1340) should contain this fix. Would be nice if you could briefly report back, if all works as expected again.

**Comment by michaeljoos72** [comment]
Updated to snapshot 1340 and all devices are online and working! Thx!

**Comment by kaikreuzer** [comment]
Excellent, thanks for testing!
